### PR TITLE
Sklearn 024

### PR DESCRIPTION
**Pull request by samihamdan**

### CODE REVIEW

**Comment by fraimondo** [comment]
Set `sklearn < 0.25` and `>= 0.22`, no?

**Comment by fraimondo** [comment]
`pandas < 1.3.0` and `pandas >= 1.1.0`? (I am using 1.1.3 in my env)

**Comment by samihamdan** [comment]
Yes I will do so immediately! 
I only wanted to check whether the bugfix is becoming green before changing anything else (np had to be fixed for that). 
There is a deprecation of `coef_` and `intercept_` in NaiveBayes, it will be removed in .26 of sklearn. 
Which leads to some warnings in julearn when testing.
Should I deal with this in this PR or should we do it in a later one ?

**Comment by fraimondo** [comment]
I would do it here. This should be the sklearn 0.24 update.

**Comment by samihamdan** [comment]
Is this fine? 
I set numpy below the next major release as well because the current one seems to work.

(Don't merge yet. I have to deal with deprecation warnings first)

**Comment by samihamdan** [comment]
From my side this PR is ready, but @fraimondo  please check whether I handled FutureWarning properly in `utils/testing.py`.

File: docs/changes/latest.inc [hunk 1]
```diff
@@ -30,5 +30,7 @@ Bugs
 
 - Fix target transformers that only normal transformers are wrapped (:gh:`94` by `Sami Hamdan`_)
 
+- Fix classification bug due to scikit-learn version 0.2.4 (:gh:`#108` by `Sami Hamdan`_)
```
Comment (fraimondo) [inline docs/changes/latest.inc:33]: sklearn version is 0.24. Was it a bug? Or something that changed within sklearn?

I would say something like "Fix compatibility with new scikit-learn release 0.24"

**Review by fraimondo — CHANGES_REQUESTED** [review]

File: docs/changes/latest.inc (hunk 1 above)
Comment (samihamdan) [reply docs/changes/latest.inc:33]: You are right this is a better way of putting it. 
We just needed an additional attribute `classes_` for classification

**Review by fraimondo — APPROVED** [review]

**Pull request merged**